<a href="https://colab.research.google.com/github/Viachok2107/Game1/blob/main/%D0%9A%D0%BE%D0%BF%D0%B8%D1%8F_%D0%B1%D0%BB%D0%BE%D0%BA%D0%BD%D0%BE%D1%82%D0%B0_%22Untitled0_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
print(4*1024*1024*8)

33554432


1,1 ответ:33554432бит

In [ ]:
print(2*1024*1024*1024*8)

17179869184


1.2,ответ:17179869184

In [ ]:
print(6291456 /1024/1024)

6.0


1.3,ответ:6

In [ ]:
2.1,ОТВЕТ:5

2.2,ОТВЕТ:3

2.3,ответ:5

3.1,ответ:1

3.2,ответ:0.650022

3.3 0.44 принимает  0.53 не принимает   0.97 общий
    

4.1  ответ: 77(16)

4.2 ответ: u

4.3 ответ: 73747564656E74(16)

5.1 ответ:110010010


5.2 ответ:  101000010

5.3 ответ: содердит ошибку т.к сообщение содержит нечетное число единиц

6.1 ответ:|000| |000| |111| |111| |111| |000| |111| |111|

6.2 ответ:01011001

6.3 Ответ:1

7.1 ответ:3

7.2 ОТВЕТ: все суждения верны

7.3 ответ:1

Тест

1.ответ:1

2.ответ:3

3.ответ:2

4.ответ:4

5.ответ:4

6.ответ:2

7.ответ:3

8.ответ:1

9.ответ:3

10.ответ:24576

In [ ]:
#8.1
print("Вячин Максим, эфбо-17-25")

Вячин Максим, эфбо-17-25


In [1]:
#8.2
x = input ("введите число")
x=float(x)
y = input ("введите число")
y=float(y)
print (x+y)

введите число0.4
введите число0.5
0.9


In [ ]:
#8.3
x = input ("введите число")
x = float (x)
y=x**5-2*x**3+1
print(y)

введите число2
17.0


In [15]:
#9.1
strange= [ [],1]
print(type(strange))

<class 'list'>


In [28]:
#9.2
for x in range (2):
  #x=float(x)
  for y in range (2):
    #y=float(y)
    if  ((x or y) and (not x or y) and not(x and y))==True:
      print("x y")
      print (x,y)

x y
0 1


In [ ]:
9.3
x = [input() for i in range(4)]
print(*x[::-1])

adada
fdfds
fdsfs
gsdsg
gsdsg fdsfs fdfds adada


In [ ]:
#10.1
import math as m
x = input ("введите число")
x = float(x)
i=m.log2(x)
print (i)


введите число5
2.321928094887362


In [16]:
#10.2
import math as m
x=input("введите число")
x=float(x)
y= input("введите число ")
y=float(y)
z=x+y
p1=x/z
p2=y/z
I=(-p1*m.log(p1,2)-p2*m.log(p2,2))
print(I)

введите число2
введите число 6
0.8112781244591328


In [ ]:
#10.3
import math as m
x = float(input("введите число"))
i = ((m.tan(m.cos(x)*m.sin(2*x))/x*m.e**x)**(m.log(x,7)))
print (i)

введите число3
1.4370551949505188


# 1. В коробке 32 карандаша, все карандаши разного цвета. Наугад вытащили красный. Какое количество информации при этом было получено?
N=32
---
32=2*i
---
i=5 бит




# 2. В коробке 50 шаров, из них 40 белых и 10 чёрных. Определить количество информации в сообщении о вытаскивании наугад белого шара и чёрного шара.
p1 = 40/50 = 0,8
---
p2 = 10/50 = 0,2
---

```
import math as m
print(m.log(1/0.8,2))
print(m.log(1/0.2,2))
---
0.3219280948873623 ≈ 0.32
2.3219280948873623 ≈ 2.32
```
i1 = log2(1/0,8)=log2(5)≈0,32бит
---
i2 = log2(1/0,2)=log2(1,25)≈2,32бит


---
Ответ:0,32бит, 2,32бит.

# 3.В озере живут караси и окуни. Подсчитано, что карасей 1500, а окуней - 500. Сколько информации содержится в сообщениях о том, что рыбак поймал карася, окуня, поймал рыбу?
1500+500=2000
p1=1500/2000=0,75
---
p2=500/2000 = 0,25
---
i1=log2(1/0,75)=0.4150374992788438 бит ≈ 0.42 бит

```
import math as m
print(m.log(1/0.75,2))
0.4150374992788438
```


---
i2=log2(1/0,25)=2бит
---
I=-p1*log(p1) - p2*log2(p2)
---
I=0,75*log2(0,75)-0,25*log2(0,25)

```
import math as m
I=(0.75*m.log(0.75,2)-0.25*m.log(0.25,2))
print(I)
0.8112781244591328 бит ≈ 0,81 бит
```
Ответ: 0.42 бит, 2 бит, 0.81 бит.

# 4. Какое количество информации несет в себе сообщение о том, что нужная вам программа находится на одной из восьми дискет?
N=8
---
8=2*i
---
i=3
---
Ответ: 3 бит


# 5. Какова мощность алфавита, с помощью которого записано сообщение, содержащее 2048 символов, если его объем составляет 1/512 часть одного мегабайта?
1/512 мб = 2 кб
---
2*1024=2024 байт=>
---
=>i=1байт=8бит
---
2**8=256
---
Ответ: 256

# 6. Книга, набранная с помощью компьютера, содержит 150 страниц; на каждой странице - 40 строк, в каждой строке - 60 символов. Каков объем информации в книге?

128 < 150 > 256
---
i=1байт
---
40*60=2400
---
2400*150=360000
---
Ответ:360000 байт


# 7. Для передачи секретного сообщения используется код, состоящий из десяти цифр. При этом все цифры кодируются одним и тем же (минимально возможным) количеством бит. Определите информационный объем сообщения длиной в 150 символов.
8 < 10 > 16
---
i=4бит
---
150*4=600
---
Ответ:600 БИТ

# 8. В кодировке Unicode на каждый символ отводится два байта. Определите информационный объем слова из двадцати четырех символов в этой кодировке.
2*24=48 БАЙТ
---
Ответ:48 Байт

# 9. В рулетке общее количество лунок равно 128. Какое количество информации мы получаем в зрительные сообщения об остановке шарика в одной из лунок?
128=2**7
---
i=7бит
---
Ответ:7бит